#4690 - Prompt default users to change their password on login

schwzr opened the pull request:
# What does this PR do?

- This PR fixes #4690.
- It provides a new feature to present a prompt at login to any of the standard users with their standard password (as defined in `StartupShutdownService:createDefaultUsers`) to change their password. 
- If we have an elevated user capable of editing other users, this user can change the standard password of other standard users as well. The new passwords can be exported as CSV.
- The attached screencast demonstrates this new feature:

![sormas-4690](https://user-images.githubusercontent.com/5512013/112333683-19874300-8cbb-11eb-9aa1-64f374ac4435.gif)

We are happy to receive feedback from your side!

Thanks to @Snorzze for providing the unit tests!

MateStrysewske commented on sormas-api/src/main/java/de/symeda/sormas/api/utils/DefaultUserHelper.java:
```diff
@@ -0,0 +1,97 @@
+		return allUsersWithDefaultPassword.contains(currentUser);
+	}
+
+	public static boolean otherUsersWithDefaultPassword(List<UserDto> allUsersWithDefaultPassword, UserDto currentUser) {
```
Rename to `otherUsersUseDefaultPassword`

MateStrysewske commented on sormas-ui/src/main/java/de/symeda/sormas/ui/login/DefaultPasswordOtherScreen.java:
```diff
@@ -0,0 +1,256 @@
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class DefaultPasswordOtherScreen extends Window {
```
Rename to `ChangeDefaultPasswordsWindow`

MateStrysewske commented on sormas-ui/src/main/java/de/symeda/sormas/ui/login/DefaultPasswordOwnScreen.java:
```diff
@@ -0,0 +1,96 @@
+import de.symeda.sormas.ui.utils.ButtonHelper;
+import de.symeda.sormas.ui.utils.CssStyles;
+
+public class DefaultPasswordOwnScreen extends Window {
```
Rename to `ChangeDefaultUserPasswordWindow`

MateStrysewske reviewed (changes requested):
Hey! Thanks for your contribution! Seems like your IDE settings aren't quite correct when it comes to import orders. java imports are supposed to be above the symeda imports. Please make sure that your IDE is set up correctly and re-format your files.

schwzr commented:
Thanks @MateStrysewske. I updated my IDE settings to not revert the import ordering of the Eclipse Code Formatting Plugin automatically and renamed the classes and method names as you requested.

MateStrysewske reviewed (approved):
Thank you! Android indeed supports some Java 8 features, but it's still using Java 7 at its core, and it especially requires external code resources (just like our API) to be written in Java 7, unfortunately.

Also, for future reference if you want to contribute code to this repository again - please make sure that every commit always references the issue number (not just the first one). That's important for us to ensure a clean commit history. I've overseen it myself when I did my initial review, so it's alright for now. :)

rzo1 commented:
@MateStrysewske Thanks for your time reviewing our pull request. For future contribution, we will adhere to the contribution guidelines. May I ask, if this will be included in the upcoming release?

MateStrysewske commented:
@rzo1 Yes, this feature will be part of 1.59.0!
